boshio/http_client.go

```diff
@@ -8,13 +8,33 @@ import (
 	"time"
 )
 
-type HTTPClient struct {
+func NewHTTPClient(host string, wait time.Duration) HTTPClient {
```

Comment:
This should be a struct return off of a function and not an interface. Concrete types should be returned by constructors.